src/features/world/ui/flowerShop/FlowerBounties.tsx

```diff
@@ -0,0 +1,254 @@
+import { useSelector } from "@xstate/react";
+import { SUNNYSIDE } from "assets/sunnyside";
+import classNames from "classnames";
+import { Button } from "components/ui/Button";
+import { Label } from "components/ui/Label";
+import { ButtonPanel, Panel } from "components/ui/Panel";
+import { CloseButtonPanel } from "features/game/components/CloseablePanel";
+import { SpeakingModal } from "features/game/components/SpeakingModal";
+import { Context } from "features/game/GameProvider";
+import { PIXEL_SCALE } from "features/game/lib/constants";
+import { weekResetsAt } from "features/game/lib/factions";
+import { MachineState } from "features/game/lib/gameMachine";
+import { getKeys } from "features/game/types/decorations";
+import { FLOWERS } from "features/game/types/flowers";
+import { BountyRequest } from "features/game/types/game";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { TimerDisplay } from "features/retreat/components/auctioneer/AuctionDetails";
+import { useAppTranslation } from "lib/i18n/useAppTranslations";
+import { NPC_WEARABLES } from "lib/npcs";
+import { useCountdown } from "lib/utils/hooks/useCountdown";
+import React, { useContext, useState } from "react";
+
+const _exchange = (state: MachineState) => state.context.state.bounties;
+
+interface Props {
+  onClose: () => void;
+}
+
+function acknowledgeIntro() {
+  localStorage.setItem(
+    "flower.bounties.acknowledged",
+    new Date().toISOString(),
+  );
+}
+
+function hasReadIntro() {
+  return !!localStorage.getItem("flower.bounties.acknowledged");
+}
+
+export const FlowerBounties: React.FC<Props> = ({ onClose }) => {
+  const { gameService } = useContext(Context);
+  const exchange = useSelector(gameService, _exchange);
+
+  const { t } = useAppTranslation();
+
+  const [showIntro, setShowIntro] = useState(!hasReadIntro());
+  const [deal, setDeal] = useState<BountyRequest>();
+
+  const state = gameService.getSnapshot().context.state;
+
+  const deals = exchange.requests.filter((deal) => deal.name in FLOWERS);
+
+  const expiresAt = useCountdown(weekResetsAt());
+
+  if (showIntro) {
+    return (
+      <SpeakingModal
+        message={[
+          {
+            text: t("bounties.flower.intro.one"),
+          },
+          {
+            text: t("bounties.flower.intro.two"),
+          },
+        ]}
+        bumpkinParts={NPC_WEARABLES.poppy}
+        onClose={() => {
+          acknowledgeIntro();
+          setShowIntro(false);
+        }}
+      />
+    );
+  }
+
+  if (deal) {
+    return (
+      <Deal deal={deal} onClose={() => setDeal(undefined)} onSold={onClose} />
+    );
```

Comment:
I found the UI closes when I complete a bounty. I think it feels better going back to the original screen:

```suggestion
      <Deal deal={deal} onClose={() => setDeal(undefined)} onSold={() => setDeal(undefined)} />
    );
```